tests/sla_tests.js

```diff
@@ -429,4 +465,180 @@ describe('SLA Bridge and Channels Tester', function() {
       }, asyncDelay);
     } 
   });
+  it('should mark a device as RINGING when dialing an outbound channel',
+      function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config='tests/testConfigs/singleEndpoint.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+    client.deviceStates.update({deviceName: 'Stasis:42',
+      deviceState: 'NOT_INUSE'}, function() {});
+
+    markAsRinging();
+    function markAsRinging() {
+      setTimeout(function() {
+        if (isMixing && bridges.length === 1 && inbound.inbound &&
+          dialed[0] && mockDeviceStates[0].hasRung &&
+          mockDeviceStates[0].deviceName === 'Stasis:42') {
+            done();
+          } else {
+            markAsRinging();
+          }
+      }, asyncDelay);
+    }
+  });
+  it('should mark a device as RINGING when dialing multiple outbound channels',
+      function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config='tests/testConfigs/multipleEndpoints.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+    client.deviceStates.update({deviceName: 'Stasis:42',
+      deviceState: 'NOT_INUSE'}, function() {});
+
+    markAsRinging();
+    function markAsRinging() {
+      setTimeout(function() {
+        if (isMixing && bridges.length === 1 && inbound.inbound &&
+          dialed[0] && dialed[1] && mockDeviceStates[0].hasRung &&
+          mockDeviceStates[0].deviceName === 'Stasis:42') {
+            done();
+        } else {
+          markAsRinging();
+        }
+      }, asyncDelay);
+    }
+  });
+  it('should give the application an invalid configuration file and promptly ' +
+      'fail out', function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    config = 'tests/testConfigs/invalid.json';
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+
+    invalidConfigurationFile();
+    function invalidConfigurationFile() {
+      setTimeout(function() {
+        if (configurationFailed) {
+          done();
+        } else {
+          invalidConfigurationFile();
+        }
+      }, asyncDelay);
+    } 
+  });
+  it('should mark a device as INUSE when dialing an outbound channel',
+      function(done) {
+    var client = getMockClient();
+    var inbound = getMockChannel();
+    inbound.inbound = true;
+    var sla = require('../lib/sla.js')(client, config, inbound, '42')
+      .catch(errHandler)
+      .done();
+    client.deviceStates.update({deviceName: 'Stasis:42',
```

Comment:
This is hard to read, you should go for one of the approached below:

client.deviceStates.update({
  deviceName: 'Stasis:42',
  deviceState: 'NOT_INUSE'
}, function() {
});

client.deviceStates.update({deviceName: 'Stasis:42',
                                            deviceState: 'NOT_INUSE'}, function() {});

The important thing is to align subsequent lines in a way that flows better and is easier to read.